### PR TITLE
Decompile reshow battle screen

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: data/data2b.s [hunk 1]
```diff
@@ -4509,7 +4510,7 @@ gUnknown_08305D0C:: @ 8305D0C
 @ 8305DCC
 	.include "data/enemy_mon_elevation.inc"
 
-@ 8305F68
+@ 8309AAC
```
Comment (Diegoisawesome) [inline data/data2b.s]: Why did you change this? It's definitely not located there.

File: src/reshow_battle_screen.c [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        SetMainCallback2(BattleMainCB2);
+        sub_805EF14();
+        break;
+    }
+
+    gBattleScripting.reshowMainState++;
+}
+
+// todo: find a better place for the struct declaration
+struct BGCntrlBitfield
```
Comment (Diegoisawesome) [inline src/reshow_battle_screen.c]: bg.h would make sense for this struct.

File: data/data2b.s (hunk 1 above)
Comment (DizzyEggg) [reply data/data2b.s]: fixed

File: src/reshow_battle_screen.c (hunk 2 above)
Comment (DizzyEggg) [reply src/reshow_battle_screen.c]: moved
